common/src/main/java/com/wynntils/screens/maps/PoiManagementScreen.java

```diff
@@ -67,79 +111,756 @@ public void onClose() {
         }
 
-        checkAvailablePages();
+        return true;
+    }
+
+    @Override
+    public boolean mouseReleased(double mouseX, double mouseY, int button) {
+        draggingScroll = false;
+        return true;
+    }
+
+    @Override
+    public boolean mouseScrolled(double mouseX, double mouseY, double deltaX, double deltaY) {
+        double scrollValue = -Math.signum(deltaY);
+        scroll((int) scrollValue);
+
+        return true;
+    }
+
+    @Override
+    public boolean mouseDragged(double mouseX, double mouseY, int button, double dragX, double dragY) {
+        if (!draggingScroll) return false;
+
+        int renderY = (int) ((this.height - backgroundHeight) / 2 + (int) (dividedHeight * 3));
+        int scrollAreaStartY = renderY + 12;
+
+        int newValue = (int) MathUtils.map(
+                (float) mouseY,
+                scrollAreaStartY,
+                scrollAreaStartY + (int) (dividedHeight * 40),
+                0,
+                Math.max(0, pois.size() - maxPoisToDisplay));
+
+        scroll(newValue - scrollOffset);
+
+        return super.mouseDragged(mouseX, mouseY, button, dragX, dragY);
+    }
+
+    private void renderScrollButton(PoseStack poseStack) {
+        // Don't render the scroll button if it will not be useable
+        if (pois.size() <= maxPoisToDisplay) return;
+
+        // Calculate where the scroll button should be on the Y axis
+        scrollButtonRenderY = (this.height - backgroundHeight) / 2
+                + (int) (dividedHeight * 3)
+                + MathUtils.map(scrollOffset, 0, pois.size() - maxPoisToDisplay, 0, scrollAreaHeight);
+
+        RenderUtils.drawScalingTexturedRect(
+                poseStack,
+                Texture.SCROLL_BUTTON.resource(),
+                scrollButtonRenderX,
+                scrollButtonRenderY,
+                1,
+                (dividedWidth / 2),
+                scrollButtonHeight,
+                Texture.SCROLL_BUTTON.width(),
+                Texture.SCROLL_BUTTON.height());
+    }
+
+    private void scroll(int delta) {
+        // Calculate how many pois should be scrolled past
+        scrollOffset = MathUtils.clamp(scrollOffset + delta, 0, Math.max(0, pois.size() - maxPoisToDisplay));
 
         populatePois();
     }
 
+    private void populatePois() {
+        // Remove old widgets
+        for (AbstractWidget widget : poiManagerWidgets) {
+            this.removeWidget(widget);
+        }
+
+        this.poiManagerWidgets.clear();
+
+        // Get full list of pois
+        pois = Managers.Feature.getFeatureInstance(MainMapFeature.class).customPois.get().stream()
+                .filter(poi -> searchMatches(poi.getName()))
+                .collect(Collectors.toList());
+
+        // No pois
+        if (pois.isEmpty()) {
+            return;
+        }
+
+        // Filter pois based on filtered icons
+        pois = pois.stream()
+                .filter(poi -> filteredIcons.getOrDefault(poi.getIcon(), true))
```

Comment:
Why true..?

Reply:
true is to include it, so the default is to include all icons. It might not need the default honestly as `updateAllUsedIcons` should hopefully populate it with all icons